# Homework 6
** Mackenna Yount **

## Question 1

I made a link to the FASTA with a simpler name:
```bash
ln -sf ncbi_dataset/data/GCF_000196035.1/GCF_000196035.1_ASM196035v1_genomic.fna listeria.fa
```

To find the size of the file I used:
```bash
ls -lh ncbi_dataset/data/GCF_000196035.1/
```
and it returned the information about the FASTA file. It is 2.8M in size.

To find the number of chromosomes in the genome:
```bash
grep -c '>' listeria.fa
```
The output was 1, so there is only one chromosome in the genome.

I looked at the first ten lines of the FASTA file to find the name(ID) and length of each chromosome in the genome:
```bash
cat listeria.fa | head 
```
In the output, it lists the name of the single chromosome as NC_003210.1. 

I then found the length of the chromosome by using the following command:
```bash
grep -v '^>' listeria.fa | tr -d '\n' | wc -c
```
The output was 2944528, so the length of the chromosome is 2,944,528 bp.

## Question 2
Generate a simulated FASTQ output for a sequencing instrument of your choice. Set the parameters so that your target coverage is 10X. 

```bash
wgsim -N 200000 -1 150 -2 100 -r 0 -R 0 -X 0 listeria.fa reads/wgsim_read1.fq reads/wgsim_read2.fq
```

I generated 200000 reads with an average read length of 150 bp. I chose these based off of Illumina's typical read lengths.

I ran the following command to see more information about the reads:
```bash
seqkit stats reads/wgsim_read1.fq reads/wgsim_read2.fq
```
And it returned the following information:
```
file                  format  type  num_seqs     sum_len  min_len  avg_len  max_len
reads/wgsim_read1.fq  FASTQ   DNA    200,000  30,000,000      150      150      150
reads/wgsim_read2.fq  FASTQ   DNA    200,000  20,000,000      100      100      100
```

How big are the FASTQ files?
```bash
echo $(du -sh reads/wgsim_read1.fq reads/wgsim_read2.fq)
```
Read 1 is 67M and read 2 is 48M.

Compress the files and report how much space it saves.
```bash
gzip reads/wgsim_read1.fq reads/wgsim_read2.fq
echo $(du -sh reads/wgsim_read1.fq.gz reads/wgsim_read2.fq.gz)
```
Read 1 is 13M and read 2 is 10M.

Could you get the same coverage with different parameter settings (read length vs read number)?

Yes, I could still get the same coverage with longer reads. For example, if I used reads with a length of 300bp (double as many aas before), I would need less readsâ€“ about 100000 of them (half as many as before) to get the same coverage.

## Question 3
How much data would be generated when covering the yeast, the drosophila, and the human genomes?

Now imagine that instead of your genome, each instrument generated reads that dont cover the yeast, the drosophila, and the human genome at 30x coverage. You don't have to run the tool!

Using the information for the genome you've studied in the previous points, for each of the organisms, estimate the size of the FASTA file that holds the genome, the number of FASTQ reads needed for 30x, and the size of the FASTQ files before and after compression.

Yeast (Saccharomyces cerevisiae):
* Size of the FASTA file: 12071326 bp
* Number of reads needed for 30x coverage (L=150bp): 800000
* Size of the FASTQ files before compression: 232M
* Size of the FASTQ files after compression: 47M

Drosophila (fruit fly):
* Size of the FASTA file: 143706478 bp
* Number of reads needed for 30x coverage (L=150bp): about 9 million
* Size of the FASTQ files before compression: 2.8G
* Size of the FASTQ files after compression: 557M

Human:
* Size of the FASTA file: about 3.2 billion bp
* Number of reads needed for 30x coverag (L=150bp): 640 million
* Size of the FASTQ files before compression: 40.8G
* Size of the FASTQ files after compression: 620M
